api/controller/impression.js

```diff
@@ -0,0 +1,63 @@
+const Bluebird = require("bluebird");
+const Controller = require("../controller/controller");
+const Impression = require("../model/impression");
+const Plan = require("../model/plan");
+const { Knex } = require('../service/database')
+
+class ImpressionController extends Controller {
+
+    getIp = (req) => {
+        return req.headers["x-forwarded-for"] || req.connection.remoteAddress;
```

Comment:
we wanted to hash the ip, no?
we don't want to keep actual client ips, plus it can save space